File: Python/demcachtachxauchiahetcho9.py
def solve(s):
    count = [0] * 10
    temp = 0
    for i in range(len(s)):
        temp = (temp + int(s[i])) % 9
        count[temp] += 1
    result = count[0]
    for i in range(9):
        result += int(count[i] * (count[i] - 1) / 2)
    return result % 1000000007

print(solve(input()))

File: Cpp/demcachtachxauchiahetcho9.cpp
#include <iostream>
#include <string>
#include <vector>
#define ll long long

const int MOD = 1000000007;
int solve(std::string s) {
    std::vector <ll> count(10, 0);
    int temp = 0;
    for (char &c : s) {
        temp = (temp + (c - '0')) % 9;
        ++count[temp];
    }
    int result = count[0];
    for (int i = 0; i < 9; ++i) result = (result + ((count[i] * (count[i] - 1)) / 2) % MOD) % MOD;
    
    return result;
}
int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    std::cout.tie(NULL);
    std::string s;
    std::cin >> s;
    std::cout << solve(s);
    return 0;
}

// code by duck it1
